libp2p/peerinfo.nim

```diff
@@ -83,6 +98,9 @@ proc publicKey*(p: PeerInfo): Option[PublicKey] {.inline.} =
     result = some(p.privateKey.getKey())
 
 proc `publicKey=`*(p: PeerInfo, key: PublicKey) =
+  if p.publicKey.isSome:
```

Comment:
can't this be part of init requirements so the setter can be removed? seems a let less error prone because suddenly there's no possibility for runtime errors..